src/Equations/SPIHandler.cpp

```diff
@@ -87,44 +109,74 @@ SPIHandler::SPIHandler(FVM::Grid *g, FVM::UnknownQuantityHandler *u, len_t *Z, l
     // Get number of grid points and number of shards
     this->nr=rGrid->GetNr();
     this->nShard=unknowns->GetUnknown(id_Yp)->NumberOfMultiples();
+    
+    this->NZ=NZ;
 
     // Memory allocation
     AllocateQuantities();
     
+    // Ablation cloud quantities
+    this->TDrift=TDrift;
```

Comment:
``this->TDrift`` is allocated in ``AllocateQuantities()``, but then immediately overwritten here, leading to a memory leak. Is this really the intended behaviour? Should ``TDrift`` really be allocated in ``AllocateQuantities()``? (if not, the ``delete [] TDrift`` should probably be moved directly to the destructor?